### PR TITLE
fix(login): don't call custom login page endpoint if the module is not installed

### PR DESCRIPTION
**Pull request by jeremyjaouen**
## Description

This PR Intends to fix an issue where the custom login_page endpoint was called even if the module was not installed

**Fixes** # MON-18113

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 21.10.x
- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [x] 23.10.x (master)

<h2> How this pull request can be tested ? </h2>

Reach Centreon Login Page with development tools > network tab open. You shouldnt see a 404 call to /it-edition-extensions/login_page

#### Community contributors & Centreon team

- [x] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: centreon/www/front_src/src/App/useApp.ts [hunk 1]
```diff
@@ -127,11 +129,16 @@ const useApp = (): UseAppState => {
...
         }
       });
-    getCustomPlatformRequest({
-      endpoint: loginPageCustomisationEndpoint
-    })
-      .then(({ platform_name }) => setPlaformName(platform_name))
-      .catch(() => undefined);
+    if (
+      !isNil(platformVersion?.modules) &&
+      !!platformVersion?.modules[`centreon-it-edition-extensions`]
```
Comment (bdauria) [inline centreon/www/front_src/src/App/useApp.ts:134]: you can use ramda's path to simplify the statement

**Comment by Med1233** [comment]
Please take into consideration that an update is required for the platformVersionsAtom atom. The values to be updated within this atom are contingent on whether the 'centreon-it-extensions' module is installed or not

for the two test cases with module installed you need to update the atom 
`    store.set(platformVersionsAtom, retrievedWebWithItEditionInstalled);
`
with : 

`const retrievedWebWithItEditionInstalled = {
  modules: {
    'centreon-it-edition-extensions': {
      fix: '0',
      major: '23',
      minor: '10',
      version: '23.10.0'
    }
  },
  web: {
    fix: '1',
    major: '21',
    minor: '10',
    version: '21.10.1'
  }
};`

**Comment by jeremyjaouen** [comment]
> Please take into consideration that an update is required for the platformVersionsAtom atom. The values to be updated within this atom are contingent on whether the 'centreon-it-extensions' module is installed or not
> 
> for the two test cases with module installed you need to update the atom ` store.set(platformVersionsAtom, retrievedWebWithItEditionInstalled);` with :
> 
> `const retrievedWebWithItEditionInstalled = { modules: { 'centreon-it-edition-extensions': { fix: '0', major: '23', minor: '10', version: '23.10.0' } }, web: { fix: '1', major: '21', minor: '10', version: '21.10.1' } };`

Was digging into this solution yesterday ! thanks for the hint ! 👍

File: centreon/www/front_src/src/Login/useLogin.ts [hunk 2]
```diff
@@ -98,14 +99,18 @@ const useLogin = (): UseLoginState => {
       suspense: false
     }
   });
-
-  const { data: loginPageCustomisationData, isLoading } =
+  const [platformVersions] = useAtom(platformVersionsAtom);
```
Comment (Thebarda) [inline centreon/www/front_src/src/Login/useLogin.ts:102]:
```suggestion
  const platformVersions = useAtomValue(platformVersionsAtom);
```

**Review by dmyios — APPROVED** [review]

**Review by Med1233 — APPROVED** [review]

**Pull request merged**